src/main/java/gov/nasa/jpl/mbee/mdk/mms/endpoints/MMSEndpoint.java

```diff
@@ -0,0 +1,111 @@
+package gov.nasa.jpl.mbee.mdk.mms.endpoints;
+
+import com.nomagic.magicdraw.core.Application;
+import com.nomagic.magicdraw.core.Project;
+import gov.nasa.jpl.mbee.mdk.http.HttpDeleteWithBody;
+import gov.nasa.jpl.mbee.mdk.mms.MMSUtils;
+import gov.nasa.jpl.mbee.mdk.util.TicketUtils;
+import org.apache.http.Consts;
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpEntityEnclosingRequest;
+import org.apache.http.client.entity.EntityBuilder;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.methods.HttpPut;
+import org.apache.http.client.methods.HttpRequestBase;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.FileEntity;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+
+public abstract class MMSEndpoint {
+    protected URIBuilder uriBuilder;
+    protected String baseUri;
+
+    public MMSEndpoint(String baseUri) {
+        try {
+            uriBuilder = new URIBuilder(baseUri);
+            this.baseUri = baseUri;
+        } catch (URISyntaxException e) {
```

Comment:
I don't think we should suppress this exception, especially considering methods in this class, e.g. `buildRequest`, throw it so usages should already be handling it.